clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2669,11 +2786,13 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     addParam(ArgTy, Kind, offsetOf(FD, ArgTy));
   }
   void addParam(QualType ArgTy, SYCLIntegrationHeader::kernel_param_kind_t Kind,
-                uint64_t OffsetAdj) {
+                uint64_t OffsetAdj, bool IsZeroOffset = false) {
```

Comment:
So I perhaps hate this for 2 reasons:
1- I hate bool parameters with a passion :) 

2- The 'offset' we add here is meaningless/not used at all anywhere, so I think we can do without the parameter/ternary in this function.  If we end up passing a non-sense offset in the ParamDesc, it doesn't matter!  It isn't usable for anything anyway.

The long and short of it here I guess is: what do you think about simplifying the code a bit by just reverting the change to addParam here?

Reply:
I assume we can remove this and just pass 0 to OffsetsetAdj when we call it. This would set the offset as CurOffset which I believe will be the size of the kernel object at this point (since it is last parameter). Is that 'clean' though?

Reply:
> I assume we can remove this and just pass 0 to `OffsetsetAdj `when we call it. This would set the offset as `CurOffset `which I believe will be the size of the kernel object at this point (since it is last parameter). I guess there is some consistency in this case because it will always be size of object. 

Actually it won't be the size of the kernel object. I was thinking of an earlier implementation of this class :) I think it will be the offset of containing class, which in this case is the kernel object itself. Which means the changes I made here is unnecessary because the offset will be 0 for kernel object! I'll verify this and revert this if I'm correct

Reply:
Reverted change